lib/events.js

```diff
@@ -42,27 +42,74 @@ exports.onData = function onData(message) {
         // and be reissued again.
       }
     );
-  } else {
-    message.del();
   }
-};
 
-exports.onError = function onError(err) {
-  logger.error('accountEvent', err);
-};
+  function primaryEmailChanged(message) {
+    var userId = message.uid.split('@')[0];
+    if (! HEX_STRING.test(userId)) {
+      message.del();
+      return logger.warn('badPrimaryEmailChanged', {userId: userId});
+    }
 
-exports.start = function start() {
-  if (! config.events.region || ! config.events.queueUrl) {
-    if (env.isProdLike()) {
-      throw new Error('config.events must be included in prod');
+    // We currently a generate a user's cache key from their
+    // credentials in a request object. This builds an object
+    // to match what the server is expecting.
+    var req = {auth: {credentials: {user: userId}}};
+    P.resolve(new Promise(function(resolve) {
+      server.methods.batch.cache.drop(req, function () {
+        logger.info('primaryEmailChanged:cacheCleared', {uid: userId});
+        notifyProfileUpdated(userId);
```

Comment:
Calling `notifyProfileUpdated` here won't cause any harm, but  don't think it's necessary - its only job is to tell the auth-server when profile data changes so that it can end push notifications to connected devices, but it in this case it already knows about the change, so we'll end up sending the push notifications twice.  Adding it here will cause us to send that push notification twice (once when auth-server processes the email change, and again when it receives this notification from profile-server)